sysml.library/Domain Libraries/Geometry/SpacialItems.sysml

```diff
@@ -0,0 +1,152 @@
+/**
+ * This package models physical items that have a spatial extent and act as a spatial frame of reference
+ * for obtaining position and displacement vectors of points within them.
+ */
+package SpatialItems {
+	private import Objects::Point;
+	private import SpatialFrames::SpatialFrame;
+	private import Shapes::Shape;
+	private import CSGShapes::CsgSolid;
+	private import CSGShapes::CsgUnion;
+	private import ShapeCalculations::ReifyShape;
+	private import SequenceFunctions::includes;
+	private import Quantities::VectorQuantityValue;
+	private import UnitsAndScales::VectorMeasurementReference;
+	private import Time::Clock;
+	private import Time::TimeInstantValue;
+	private import ISQ::Cartesian3dSpatialCoordinateSystem;
+	private import ControlFunctions::forAll;
+	private import ControlFunctions::reduce;
+	
+	/**
+	 * A SpatialItem is an Item with a three dimensional spatial extent that also acts as a SpacialFrame of reference.
+	 */
+	item def SpatialItem :> SpatialFrame {
+		/**
+		 * A local Clock to be used as the corresponding time reference within this SpatialItem
+		 * and any componentItems. By default this is the global defaultClock.
+		 */
+		item localClock : Clock[1] default Time::defaultClock;
+		
+		/**
+		 * The VectorMeasurementReference to be used as the measurement reference for position and displacement
+		 * vector values relative to this SpatialItem.
+		 */
+		attribute coordinateSystem : VectorMeasurementReference[1] default Cartesian3dSpatialCoordinateSystem() {
+			attribute :>> dimensions = 3;
```

Comment:
On new usages we should use `coordinateFrame`. I will raise a JIRA issue to rename all uses of CoordinateSystem in the libraries.

Reply:
Updated.